src/main/java/teammates/ui/controller/ControllerServlet.java

```diff
@@ -37,6 +37,7 @@
 public class ControllerServlet extends HttpServlet {
 
     private static final Logger log = Logger.getLogger();
+    UserType userType = new GateKeeper().getCurrentUser();
```

Comment:
I believe this needs not be a class variable. Just define it somewhere on the method body of `doPost`?